crates/oxc_ast/src/ast/jsx.rs

```diff
@@ -165,12 +168,14 @@ pub struct JSXClosingFragment {
 #[cfg_attr(feature = "serialize", derive(Serialize, Tsify))]
 #[serde(untagged)]
 pub enum JSXElementName<'a> {
+    /// `<div />`
+    Identifier(Box<'a, JSXIdentifier<'a>>) = 1,
     /// `<Apple />`
-    Identifier(Box<'a, JSXIdentifier<'a>>) = 0,
+    IdentifierReference(Box<'a, JSXIdentifierReference<'a>>) = 0,
```

Comment:
Yes, but as I replied below that:

> I think we can use `IdentifierReference` and `IdentifierName` directly and get ESTree compatibility with a custom `Serialize` impl.

In my view it'll simplify things a lot if we have only one type which can be a reference - `IdentififerReference`. Then we don't have to remember everywhere to also check for `JSXIdentifierReference`s - we're bound to forget that in some places, which would lead to bugs.

I'd be happy to do the `Serialize` + `Tsify` impls to make sure it comes out correctly in JSON AST. I've done a bunch of the other ones for cases where our internal types don't align with ESTree. It was hard to figure that out at first, but now I got the hang of it...

Reply:
> In my view it'll simplify things a lot if we have only one type which can be a reference - `IdentififerReference`. Then we don't have to remember everywhere to also check for `JSXIdentifierReference`s - we're bound to forget that in some places, which would lead to bugs.
> 

Yes, This is also what I am concerned

> I'd be happy to do the `Serialize` + `Tsify` impls to make sure it comes out correctly in JSON AST. I've done a bunch of the other ones for cases where our internal types don't align with ESTree. It was hard to figure that out at first, but now I got the hang of it...

Great! I have no opinion on this, it is up to @Boshen